### ISSUE TITLE
Regression on font sizing on neovide 0.12.0

### ISSUE THREAD

**Issue by borisfaure** [neovide/neovide#2214]
**Describe the bug**
Font size in font settings `set guifont=PragmataPro\ Mono\ Liga:h10` not taken into account with neovide-0.12.0

**To Reproduce**
1. start neovide

**Expected behavior**
The font appears large and any change to the guifont parameter is not taken into account. This worked fine on 0.11.2

**Screenshots**
.

**Desktop (please complete the following information):**
 - OS: Gentoo Linux
 - Neovide Version 0.12.0 (or 5743356 that introduced this regression)
 - Neovim Version 0.9.4

**Please run `neovide --log` and paste the contents of the `.log` file created in the current directory here:**
The file is too large, but here is `grep -i font neovide_rCURRENT.log` 
```
54:TRACE [neovide::channel_utils] neovide::editor::EditorCommand NeovimRedrawEvent(OptionSet { gui_option: GuiFont("") })
55:TRACE [neovide::channel_utils] neovide::editor::EditorCommand NeovimRedrawEvent(OptionSet { gui_option: GuiFontWide("") })
67:TRACE [neovide::editor] Option set GuiFont("")
69:TRACE [neovide::editor] Option set GuiFontWide("")
90:TRACE [neovide::channel_utils] alloc::vec::Vec<neovide::renderer::DrawCommand> [FontChanged(""), LineSpaceChanged(0), DefaultStyleChanged(Style { colors: Colors { foreground: Some(Color4f { r: 1.0, g: 1.0, b: 1.0, a: 1.0 }), background: Some(Color4f { r: 0.0, g: 0.0, b: 0.0, a: 1.0 }), special: Some(Color4f { r: 1.0, g: 0.0, b: 0.0, a: 1.0 }) }, reverse: false, italic: false, bold: false, strikethrough: false, blend: 0, underline: None })]
296:TRACE [neovide::channel_utils] neovide::editor::EditorCommand NeovimRedrawEvent(OptionSet { gui_option: GuiFont("PragmataPro Mono Liga:h10") })
323:TRACE [neovide::editor] Option set GuiFont("PragmataPro Mono Liga:h10")
398:TRACE [neovide::channel_utils] alloc::vec::Vec<neovide::renderer::DrawCommand> [LineSpaceChanged(0), Window { grid_id: 1, command: Clear }, Window { grid_id: 1, command: DrawLine { row: 49, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 48, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 47, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 46, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 45, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 44, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 43, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 42, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 41, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 40, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 39, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 38, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 37, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 36, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 35, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 34, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 33, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 32, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 31, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 30, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 29, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 28, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 27, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 26, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 25, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 24, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 23, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 22, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 21, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 20, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 19, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 18, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 17, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 16, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 15, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 14, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 13, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 12, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 11, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 10, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 9, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 8, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 7, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 6, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 5, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 4, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 3, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 2, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 1, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 0, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, FontChanged("PragmataPro Mono Liga:h10"), Window { grid_id: 1, command: Clear }, Window { grid_id: 1, command: DrawLine { row: 49, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 48, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 47, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 46, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 45, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 44, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 43, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 42, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 41, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 40, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 39, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 38, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 37, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 36, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 35, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 34, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 33, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 32, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 31, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 30, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 29, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 28, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 27, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 26, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 25, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 24, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 23, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 22, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 21, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 20, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 19, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 18, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 17, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 16, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 15, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 14, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 13, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 12, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 11, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 10, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 9, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 8, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 7, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 6, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 5, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 4, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 3, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 2, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 1, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, Window { grid_id: 1, command: DrawLine { row: 0, line_fragments: [LineFragment { text: "                                                                                                    ", window_left: 0, width: 100, style: None }] } }, ModeChanged(Normal), UpdateCursor(Cursor { grid_position: (0, 0), parent_window_id: 1, shape: Block, cell_percentage: Some(0.0), blinkwait: Some(0), blinkon: Some(0), blinkoff: Some(0), style: None, enabled: true, double_width: false, grid_cell: (" ", None) })]
518:DEBUG [neovide::renderer::fonts::caching_shaper] Original font_size: 31.11px
519:TRACE [neovide::renderer::fonts::font_loader] Loading font FontKey { bold: false, italic: false, family_name: None, hinting: Full, edging: AntiAlias }
520:DEBUG [neovide::renderer::fonts::caching_shaper] Original font_width: 19.15px
521:DEBUG [neovide::renderer::fonts::caching_shaper] Font width: 19.15px (avg: 19.15px)
522:DEBUG [neovide::renderer::fonts::caching_shaper] Fudge factor: 0.99
523:DEBUG [neovide::renderer::fonts::caching_shaper] Fudged font size: 30.88px
524:DEBUG [neovide::renderer::fonts::caching_shaper] Fudged font width: 19.15px
525:TRACE [neovide::renderer::fonts::font_loader] Loading font FontKey { bold: false, italic: false, family_name: None, hinting: Full, edging: AntiAlias }
526:INFO [neovide::window::window_wrapper] window created (scale_factor: 1.6667, font_dimensions: Dimensions { width: 19, height: 39 })
528:DEBUG [neovide::renderer::fonts::caching_shaper] Updating font:
529:TRACE [neovide::renderer::grid_renderer] Updated font dimensions: Dimensions { width: 19, height: 39 }
530:DEBUG [neovide::renderer::fonts::caching_shaper] Updating linespace: 0
531:DEBUG [neovide::renderer::fonts::caching_shaper] Linespace updated to: 0
532:DEBUG [neovide::renderer::fonts::caching_shaper] Original font_size: 31.11px
533:TRACE [neovide::renderer::fonts::font_loader] Loading font FontKey { bold: false, italic: false, family_name: None, hinting: Full, edging: AntiAlias }
534:DEBUG [neovide::renderer::fonts::caching_shaper] Original font_width: 19.15px
535:DEBUG [neovide::renderer::fonts::caching_shaper] Font width: 19.15px (avg: 19.15px)
536:DEBUG [neovide::renderer::fonts::caching_shaper] Fudge factor: 0.99
537:DEBUG [neovide::renderer::fonts::caching_shaper] Fudged font size: 30.88px
538:DEBUG [neovide::renderer::fonts::caching_shaper] Fudged font width: 19.15px
539:TRACE [neovide::renderer::grid_renderer] Updated font dimensions: Dimensions { width: 19, height: 39 }
540:DEBUG [neovide::renderer::fonts::caching_shaper] Updating linespace: 0
541:DEBUG [neovide::renderer::fonts::caching_shaper] Linespace updated to: 0
542:DEBUG [neovide::renderer::fonts::caching_shaper] Original font_size: 31.11px
543:TRACE [neovide::renderer::fonts::font_loader] Loading font FontKey { bold: false, italic: false, family_name: None, hinting: Full, edging: AntiAlias }
544:DEBUG [neovide::renderer::fonts::caching_shaper] Original font_width: 19.15px
545:DEBUG [neovide::renderer::fonts::caching_shaper] Font width: 19.15px (avg: 19.15px)
546:DEBUG [neovide::renderer::fonts::caching_shaper] Fudge factor: 0.99
547:DEBUG [neovide::renderer::fonts::caching_shaper] Fudged font size: 30.88px
548:DEBUG [neovide::renderer::fonts::caching_shaper] Fudged font width: 19.15px
549:TRACE [neovide::renderer::grid_renderer] Updated font dimensions: Dimensions { width: 19, height: 39 }
550:DEBUG [neovide::renderer::fonts::caching_shaper] Updating font: PragmataPro Mono Liga:h10
551:TRACE [neovide::renderer::fonts::font_loader] Loading font FontKey { bold: false, italic: false, family_name: Some("PragmataPro Mono Liga"), hinting: Full, edging: AntiAlias }
552:DEBUG [neovide::renderer::fonts::caching_shaper] Font updated to: PragmataPro Mono Liga:h10
553:DEBUG [neovide::renderer::fonts::caching_shaper] Original font_size: 22.22px
554:TRACE [neovide::renderer::fonts::font_loader] Loading font FontKey { bold: false, italic: false, family_name: Some("PragmataPro Mono Liga"), hinting: Full, edging: AntiAlias }
555:DEBUG [neovide::renderer::fonts::caching_shaper] Original font_width: 11.11px
556:DEBUG [neovide::renderer::fonts::caching_shaper] Font width: 11.11px (avg: 11.11px)
557:DEBUG [neovide::renderer::fonts::caching_shaper] Fudge factor: 0.99
558:DEBUG [neovide::renderer::fonts::caching_shaper] Fudged font size: 22.00px
559:DEBUG [neovide::renderer::fonts::caching_shaper] Fudged font width: 11.11px
560:TRACE [neovide::renderer::grid_renderer] Updated font dimensions: Dimensions { width: 11, height: 25 }
568:TRACE [neovide::renderer::fonts::caching_shaper] Shaping text: 1
569:TRACE [neovide::renderer::fonts::font_loader] Loading font FontKey { bold: false, italic: false, family_name: Some("PragmataPro Mono Liga"), hinting: Full, edging: AntiAlias }
673:TRACE [neovide::renderer::fonts::caching_shaper] Shaping text: ~
674:TRACE [neovide::renderer::fonts::caching_shaper] Shaping text: NVIM v0.9.4
675:TRACE [neovide::renderer::fonts::caching_shaper] Shaping text: Nvim is open source and freely distributable
676:TRACE [neovide::renderer::fonts::caching_shaper] Shaping text: https://neovim.io/#chat
677:TRACE [neovide::renderer::fonts::caching_shaper] Shaping text: type  :help nvim
678:TRACE [neovide::renderer::fonts::caching_shaper] Shaping text: <Enter>
679:TRACE [neovide::renderer::fonts::caching_shaper] Shaping text: if you are new!
680:TRACE [neovide::renderer::fonts::caching_shaper] Shaping text: type  :checkhealth
681:TRACE [neovide::renderer::fonts::caching_shaper] Shaping text: to optimize Nvim
682:TRACE [neovide::renderer::fonts::caching_shaper] Shaping text: type  :q
683:TRACE [neovide::renderer::fonts::caching_shaper] Shaping text: to exit
684:TRACE [neovide::renderer::fonts::caching_shaper] Shaping text: type  :help
685:TRACE [neovide::renderer::fonts::caching_shaper] Shaping text: for help
686:TRACE [neovide::renderer::fonts::caching_shaper] Shaping text: type  :help news
687:TRACE [neovide::renderer::fonts::caching_shaper] Shaping text: to see changes in v0.9
688:TRACE [neovide::renderer::fonts::caching_shaper] Shaping text: Help poor children in Uganda!
689:TRACE [neovide::renderer::fonts::caching_shaper] Shaping text: type  :help iccf
690:TRACE [neovide::renderer::fonts::caching_shaper] Shaping text: for information
691:TRACE [neovide::renderer::fonts::caching_shaper] Shaping text: "[No Name]" --No lines in buffer--
692:TRACE [neovide::renderer::fonts::caching_shaper] Shaping text: 1|0,0|All|[No Name]                                                              0x0|none|utf-8|unix
693:TRACE [neovide::renderer::fonts::font_loader] Loading font FontKey { bold: true, italic: false, family_name: Some("PragmataPro Mono Liga"), hinting: Full, edging: AntiAlias }
694:TRACE [neovide::renderer::fonts::caching_shaper] Shaping text:
697:TRACE [neovide::renderer::fonts::caching_shaper] Shaping text: 1|0,0|All|[No Name]                                                                                                                                        0x0|none|utf-8|unix
841:TRACE [neovide::renderer::fonts::caching_shaper] Shaping text: :
915:TRACE [neovide::renderer::fonts::caching_shaper] Shaping text: :q
970:TRACE [neovide::renderer::fonts::caching_shaper] Shaping text: :qa
```

**Additional context**
I ran `git bisect` to find the commit that introduced this regression but found out that it's a very large squashed commit: https://github.com/neovide/neovide/commit/5743356d654dbeaf213c0da3e2988c4d89c218e1
Please do not squash commits since it makes finding regressions way harder and may hinder willingness to contribute.

**Comment by fredizzimo** [comment]
The font sizes work for me, and for many other people, that have been testing this before the release. In fact we should now report errors when the font fails to load, with this pr https://github.com/neovide/neovide/pull/2089. I also don’t immediately see something wrong in the logs, but I will take a closer look later.

I’m not particularly fond of squashing commits either, but it does not hinder bisecting. Once the squashed commit is found you can check out the PR code and continue bisecting from there.

In this case it was simple, there’s exactly one commit that touches the caching shaper and could cause the problem, this one https://github.com/neovide/neovide/commit/1fc7d38634f364bd04a7d4fc8a2a31862da20e1f

But if reverting that, fixes the issue, then there’s probably some underlying issue that was masked before.

Or possibly a merge error in this commit https://github.com/neovide/neovide/commit/d42369c4984a50eb2fb2b37404b8e9f7a469c172#diff-ab22eb44387b5cd97300475e15e02e47e8725088ccbfeebc7fe99cac0a8d9bc2

**Comment by fredizzimo** [comment]
Actually, I do see a problem, I’m not sure why it hasn’t been noticed before though

https://github.com/neovide/neovide/blob/af9869bd6d8d3434b187e29f85032ee74e40b536/src/renderer/fonts/caching_shaper.rs#L170-L173

The debug statements should be moved down to after `FontLoader::new`, and there should be another `self.font_info = None` before that.

I won’t have a acess to a developer environment for three weeks, so it would be good if someone else, could test and fix this.

**Comment by borisfaure** [comment]
Thanks. I think I have some good pointers to try working on this. Enjoy your time off.

**Comment by rawlingsr** [comment]
For me, Neovide Just crashes if one or more of my fallback fonts aren't found since this update.  It says "Press ENTER or type command to continue" but is completely locked, accepting no keyboard input and requiring me to force kill the process manually. Here's the value I used to use for guifont:

```
opt.guifont = 'Source Code Pro Nerd Font:12, Fira Code:12, Source Code Pro:12, Consolas:12'
```

I can't say for sure that this was being parsed exactly correctly prior to the update, or that the syntax I was using was correct, but I do know that one of the fonts on my list was used instead of the system default. I changed it to this, which I am more confident should have been correct:

```
opt.guifont = 'SauceCodePro_Nerd_Font,Fira_Code,Source_Code_Pro,Consolas,Mono:h12'
```

But the error persisted. And here's what I had to change it to on arch linux, with consolas commented out so that I don't get an error that specifically mentions that font not being found.:

```
opt.guifont = {
    'SauceCodePro_Nerd_Font',
    'Fira_Code',
    'Source_Code_Pro',
    -- 'Consolas',
    ':h12'
}
```
From what I can see there are several problems that have been introduced, at least when lua is used for the config files, and I haven't checked if they were all introduced with https://github.com/neovide/neovide/pull/2089

1. guifont settings that had been behaving fine(whether or not they were malformed) now don't behave fine.
2. Fallback fonts aren't functioning as fallbacks, an error is thrown if any named font isn't found.
3. Neovide hard locks instead of gracefully using the next listed font.

**Comment by fredizzimo** [comment]
@borisfaure, considering that my suggested fix did not work, and that you had to divide by the scale factor, i wonder if the scale factor is the actual issue for you.

If you are on X11, winit will set a strange scale factor for you
* https://github.com/rust-windowing/winit/issues/2231

But you can override it with the environment variable `WINIT_X11_SCALE_FACTOR=1`. Wayland also works as expected.

**Comment by rawlingsr** [comment]
@fredizzimo I think it would be completely reasonable to let a user define when they're warned about missing fallback fonts. Early on in writing their config files, a user might want to know if there's a typo, or unexpected difference between the font names they know, and what Neovide expects. Later though, the same user might be synchronizing their config files across different computers, where installing all of the exact same fonts with the same names would be unreasonable.

In other words, let the user choose between perhaps the following conditions:
- None of the listed fonts are found
- At least the first listed font is not found
- Any listed font is not found

And then also complain if a value is passed to guifont, but neovide can't parse it without just "guessing" what the user meant.

**Comment by fredizzimo** [comment]
@borisfaure, did the `WINIT_X11_SCALE_FACTOR` solve your issue? I created this PR https://github.com/neovide/neovide/pull/2308, with the correct fix for the issue I reported, but it probably won't help for you. It turned out that it didn't cause any real problems in practice, that I was able to see at least.

@rawlingsr, That's one of the main reasons why we moved the font configuration to the config file https://neovide.dev/config-file.html, which is machine specific, and runs on the same machine as Neovide. And if that does not work for you, you can always add an if statement in your neovim init file using [hostname()](https://neovim.io/doc/user/builtin.html#hostname()), or in lua [uv.os.gethostname()](https://neovim.io/doc/user/luvref.html#uv.os_gethostname()), or use some of the other functions that provide information about the system.

Ignoring errors is almost always a bad idea, and often leads to unnecessary trouble shooting and even bug reports. The code also gets complex, especially if you have a choice of what type of errors you ignore.

**Comment by borisfaure** [comment]
@fredizzimo It's way better with `WINIT_X11_SCALE_FACTOR` . The font rendering is not perfect though.

**Comment by fredizzimo** [comment]
@borisfaure, could you take a screenshot, so I can see what you mean? If it shows up in the screenshot.

**Comment by borisfaure** [comment]
This is neovide with `set guifont=PragmataPro\ Mono\ Liga:h18`:
![neovide_blurry](https://github.com/neovide/neovide/assets/27988/17f889c7-7f2d-4794-9bff-278d7c9d0f39)
This is Terminology rendering the same font:
![pixel-perfect](https://github.com/neovide/neovide/assets/27988/e680c19e-3324-41b0-bc55-90d1767da180)

Now, what is strange, is neovide with `set guifont=PragmataPro\ Mono:h18`:
![neovide_fraktur](https://github.com/neovide/neovide/assets/27988/5eac1f91-ea73-4fdd-b4c7-aa6baf18be70)
I have not dug into it yet.

**Comment by fredizzimo** [comment]
The font sizes should now be the same as most other applications. A fix where the config file font size was taking pixels instead of points was also fixed. And on X11, you need to specify the scale factor yourself. https://neovide.dev/faq.html#font-size-is-weird-with-high-dpi-display-on-x11

**Comment by borisfaure** [comment]
I consider this is fixed. Thanks!
